tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,635 @@
+
+
+def test_computation_output_1():
+    """Test output if entry contains no dataframes/series"""
+    data = {"x": range(1, 4), "y": [1, 2]}
+    expected = pd.DataFrame({
+        'x': {
+            0: 1,
+            1: 1,
+            2: 2,
+            3: 2,
+            4: 3,
+            5: 3
+        },
+        'y': {
+            0: 1,
+            1: 2,
+            2: 1,
+            3: 2,
+            4: 1,
+            5: 2
+        }
+    })
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_computation_output_2():
+    """Test output if entry contains only dataframes/series"""
+    data = {
+        "df": pd.DataFrame({
+            "x": range(1, 6),
+            "y": [5, 4, 3, 2, 1]
+        }),
+        "df1": pd.DataFrame({
+            "x": range(4, 7),
+            "y": [6, 5, 4]
+        })
+    }
+
+    expected = pd.DataFrame([{
+        'df_x': 1,
+        'df_y': 5,
+        'df1_x': 4,
+        'df1_y': 6
+    }, {
+        'df_x': 1,
+        'df_y': 5,
+        'df1_x': 5,
+        'df1_y': 5
+    }, {
+        'df_x': 1,
+        'df_y': 5,
+        'df1_x': 6,
+        'df1_y': 4
+    }, {
+        'df_x': 2,
+        'df_y': 4,
+        'df1_x': 4,
+        'df1_y': 6
+    }, {
+        'df_x': 2,
+        'df_y': 4,
+        'df1_x': 5,
+        'df1_y': 5
+    }, {
+        'df_x': 2,
+        'df_y': 4,
+        'df1_x': 6,
+        'df1_y': 4
+    }, {
+        'df_x': 3,
+        'df_y': 3,
+        'df1_x': 4,
+        'df1_y': 6
+    }, {
+        'df_x': 3,
+        'df_y': 3,
+        'df1_x': 5,
+        'df1_y': 5
+    }, {
+        'df_x': 3,
+        'df_y': 3,
+        'df1_x': 6,
+        'df1_y': 4
+    }, {
+        'df_x': 4,
+        'df_y': 2,
+        'df1_x': 4,
+        'df1_y': 6
+    }, {
+        'df_x': 4,
+        'df_y': 2,
+        'df1_x': 5,
+        'df1_y': 5
+    }, {
+        'df_x': 4,
+        'df_y': 2,
+        'df1_x': 6,
+        'df1_y': 4
+    }, {
+        'df_x': 5,
+        'df_y': 1,
+        'df1_x': 4,
+        'df1_y': 6
+    }, {
+        'df_x': 5,
+        'df_y': 1,
+        'df1_x': 5,
+        'df1_y': 5
+    }, {
+        'df_x': 5,
+        'df_y': 1,
+        'df1_x': 6,
+        'df1_y': 4
+    }])
+
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_computation_output_3():
+    """Test mix of dataframes and lists"""
+    data = {
+        "df": pd.DataFrame({
+            "x": range(1, 3),
+            "y": [2, 1]
+        }),
+        "z": range(1, 4)
+    }
+    expected = pd.DataFrame([{
+        'df_x': 1,
+        'df_y': 2,
+        'z': 1
+    }, {
+        'df_x': 1,
+        'df_y': 2,
+        'z': 2
+    }, {
+        'df_x': 1,
+        'df_y': 2,
+        'z': 3
+    }, {
+        'df_x': 2,
+        'df_y': 1,
+        'z': 1
+    }, {
+        'df_x': 2,
+        'df_y': 1,
+        'z': 2
+    }, {
+        'df_x': 2,
+        'df_y': 1,
+        'z': 3
+    }])
```

Comment:
This possibly could be shortened:

```suggestion
    expected = pd.DataFrame(
        {
            "df_x": [1, 1, 1, 2, 2, 2,],
            "df_y": [2, 2, 2, 1, 1, 1,],
            "z": [1, 2, 3, 1, 2, 3,],
        }
    )
```

Could you apply the same style of changes to the other examples that are verbose in the same fashion? I'll copy/paste the following comment so you have a pseudo-checklist.

Please change this to column-style DataFrame dictionary-based construction.